app/assets/javascripts/models/story.js

```diff
@@ -62,55 +62,31 @@ var Story = module.exports = Backbone.Model.extend({
     model.setColumn();
   },
 
-  moveBetween: function(before, after) {
-    var beforeStory = this.collection.get(before);
-    var afterStory = this.collection.get(after);
-    var difference = (afterStory.position() - beforeStory.position()) / 2;
-    var newPosition = difference + beforeStory.position();
-    this.set({position: newPosition});
-    this.collection.sort();
+  move: function(previous_story_id, next_story_id) {
+    var newPosition = this.collection.calculateNewPosition(previous_story_id, next_story_id)
+    this.setPosition(newPosition);
     return this;
   },
 
-  moveAfter: function(beforeId) {
-    var before = this.collection.get(beforeId);
-    var after = this.collection.nextOnColumn(before);
-    var afterPosition;
-    if (typeof after === 'undefined') {
-      afterPosition = before.position() + 2;
-    } else {
-      afterPosition = after.position();
+  setPosition: function(position) {
+    this.set({position: position});
+    if(this.positionDecimalPlacesOverflow()){
+      this.collection.normalizePositions(this.column);
     }
-    var difference = (afterPosition - before.position()) / 2;
-    var newPosition = difference + before.position();
-    this.set({position: newPosition});
-    this.saveSorting();
-    return this;
   },
 
-  moveBefore: function(afterId) {
-    var after = this.collection.get(afterId);
-    var before = this.collection.previousOnColumn(after);
-    var beforePosition;
-    if (typeof before === 'undefined') {
-      beforePosition = 0.0;
-    } else {
-      beforePosition = before.position();
+  positionDecimalPlacesOverflow: function() {
+    const DECIMAL_PLACES_LIMIT = 10;
```

Comment:
Can we put it in the top of the file? I think it'll be easier to look for it in the future.